ddtrace/bootstrap/sitecustomize.py

```diff
@@ -74,6 +69,88 @@ def update_patched_modules():
         EXTRA_PATCHED_MODULES[module] = asbool(should_patch)
 
 
+if PY2:
+    _unloaded_modules = []
+
+
+def gevent_is_installed():
+    # https://stackoverflow.com/a/51491863/735204
+    if sys.version_info >= (3, 4):
+        return importlib.util.find_spec("gevent")
+    elif sys.version_info >= (3, 3):
+        return importlib.find_loader("gevent")
+    elif sys.version_info >= (3, 1):
+        return importlib.find_module("gevent")
+    elif sys.version_info >= (2, 7):
+        try:
+            imp.find_module("gevent")
+        except ImportError:
+            return False
+        else:
+            return True
+    return False
+
+
+def should_cleanup_loaded_modules():
+    dd_unload_sitecustomize_modules = os.getenv("DD_UNLOAD_MODULES_FROM_SITECUSTOMIZE", default="auto").lower()
+    if dd_unload_sitecustomize_modules == "0":
+        log.debug("skipping sitecustomize module unload because of DD_UNLOAD_MODULES_FROM_SITECUSTOMIZE == 0")
+        return False
+    elif dd_unload_sitecustomize_modules not in ("1", "auto"):
+        log.debug(
+            "skipping sitecustomize module unload because of invalid envvar value"
+            "DD_UNLOAD_MODULES_FROM_SITECUSTOMIZE == {}",
+            dd_unload_sitecustomize_modules,
+        )
+        return False
+    elif dd_unload_sitecustomize_modules == "auto" and not gevent_is_installed():
```

Comment:
The way it's written right now, `cleanup_loaded_modules` includes the decision of whether to actually do the cleanup. I think this encapsulation makes the main logic of this file a bit easier to follow. Maybe `cleanup_loaded_modules` should get a name that more clearly indicates that it's deciding whether to noop. What do you think @Yun-Kim?

Reply:
I was commenting more in regards to the logic of how we decide to return True in this function `should_cleanup_loaded_modules`, if we find that the logs are important than I'm happy with this as is, but it would probably be cleaner to just do something like:
```python
def should_cleanup_loaded_modules():
    dd_unload_sitecustomize_modules = os.getenv("DD_UNLOAD_MODULES_FROM_SITECUSTOMIZE", default="auto").lower()
    return dd_unload_sitecustomize_modules == "auto" and any(
        is_installed(module_name) for module_name in MODULES_THAT_TRIGGER_CLEANUP_WHEN_INSTALLED
    )
```

Reply:
On the top of encapsulating this logic, personally I think it would make more sense (more readable too?) to have the logic of deciding if we call `cleanup_loaded_modules()` to outside the function itself, like so:
```python
if should_cleanup_loaded_modules():
    cleanup_loaded_modules()
```
But I'm open to any counterarguments 😄

Reply:
Because there are two calls to `cleanup_loaded_modules`, I think there's actually a slight maintainability hit to the approach of putting the `should` check outside of that function. Future adjustments would need to be sure to change it in both places. I'll split the difference and make the name more descriptive.